src/viam/sdk/registry/registry.hpp

```diff
@@ -58,6 +58,36 @@ class ResourceRegistration {
     const google::protobuf::ServiceDescriptor* service_descriptor_;
 };
 
+/// @brief subclass of ResourceRegistration with templatized defaults
+template <typename MyResourceClient,
+          typename MyResourceServer,
+          typename MyProtoService,
+          typename MyResourceReg>
+class ResourceRegistration2 : public ResourceRegistration {
+   public:
+    using ResourceRegistration::ResourceRegistration;
+    std::shared_ptr<ResourceServer> create_resource_server(
+        std::shared_ptr<ResourceManager> manager) {
+        return std::make_shared<MyResourceServer>(manager);
+    }
+
+    std::shared_ptr<Resource> create_rpc_client(std::string name,
+                                                std::shared_ptr<grpc::Channel> chan) {
+        return std::make_shared<MyResourceClient>(std::move(name), std::move(chan));
+    }
+
+    static std::shared_ptr<ResourceRegistration> resource_registration() {
+        const google::protobuf::DescriptorPool* p =
+            google::protobuf::DescriptorPool::generated_pool();
+        const google::protobuf::ServiceDescriptor* sd =
+            p->FindServiceByName(MyProtoService::service_full_name());
+        if (!sd) {
+            throw std::runtime_error("Unable to get service descriptor");
+        }
+        return std::make_shared<MyResourceReg>(sd);
```

Comment:
I wonder if you could get rid of the CRTP if this wasn't a static member of the template, but a free function. Something like:

```
template <typename ClientType, typename ServerType, typename ProtoServiceType>
std::shared_ptr<ResourceRegistration> make_resource_registration() {
    // ...
    return std::make_shared<ResourceRegistration2<ClientType, ServerType, ProtoServiceType>>(sd);
}
```

Or perhaps it could be a member function of `Registry`:

```
class Registry {
public:
...
template<typename T, typename TClient, typename TServer>
void register_resource() { ... };
```

And then those init blocks would look like:

```
namespace {
bool init() {
    Registry::register_resource<MovementSensor, MovementSensorClient, MovementSensorServer>();
    return true;
};

// NOLINTNEXTLINE(cert-err58-cpp)
const bool inited = init();
}  // namespace